### ISSUE TITLE
Consider using experiment friendly name when using `Report Issue` command

### ISSUE THREAD

**Issue by karrtikr** [microsoft/vscode#194330]
"A/B experiments" under issues: https://github.com/microsoft/vscode-python/issues/22100

![e677f3e4-58ee-417c-b885-942df7ad464c](https://github.com/microsoft/vscode/assets/13199757/05bafb33-4467-46f1-b600-d7e74a86f8af)

don't seem very useful. They don't seem to contain the user friendly experiment name that I can recognize, rather an encoded one.

We should be able to change that to the feature gate name, the one thing is that this is more specific in the sense that it maps to a specific stage (e.g. public or insider) and step (10% or 20% etc). cc/ @luabud

**Comment by TylerLeonhardt** [comment]
Do we want to expose experiment names though?

**Comment by karrtikr** [comment]
`vscode-tas-client` requires name of experiment https://github.com/microsoft/vscode/blob/a6403e4552d39c1de87a24ad953d7d33be08cd74/extensions/typescript-language-features/src/experimentationService.ts#L27

so I would assume name of the experiment is already publicly available within this repository. Is that not the case?

**Comment by TylerLeonhardt** [comment]
fair enough... I guess the other issue is that for some users, the issue reporter opens issues in the browser via github's `body=` query parameter. This parameter has a limit to the number of characters that can fit in the parameter. If we go over, then the user needs to paste the contents into the issue... which... users forget all to often to do that.

My concern is that the names are of variable length and will cause the payload to the `body` query parameter to be too large and cause us to miss getting the important info we need to fix the user's issue.

**Comment by karrtikr** [comment]
That's true, maybe we can fallback to using encoded name if it's more than the limit. But I would assume in most cases it's pretty close, for example:
`"pythonTerminalEnvVarActivation"`
v/s
`"pythonnosmt12:30797651"`

**Comment by luabud** [comment]
Hmm that's a good point. In the past we had a limit in control tower of 50 characters for feature name, but it seems we don't have a limit anymore. Most of our feature names should be fairly short, except for the ones based on setting names.

**Comment by TylerLeonhardt** [comment]
Sounds good. We can probably check to see if the limit is hit with the names and if so, fall back to the ids if that fits.
